bin/install_swiftpm_dependencies

```diff
@@ -37,29 +98,34 @@ sudo defaults write com.apple.dt.Xcode IDEPackageSupportUseBuiltinSCM YES
 add_host_to_ssh_known_hosts bitbucket.org
 add_host_to_ssh_known_hosts github.com
 
-# Based on the project type, resolve the packages using the correct build type
-if [[ "$BUILD_TYPE" == "XCODEBUILD" ]]; then
-  echo "	Resolving packages with \`xcodebuild\`"
-  xcodebuild -resolvePackageDependencies -onlyUsePackageVersionsFromResolvedFile
-elif [[ "$BUILD_TYPE" == "SWIFT" ]]; then
-  echo "	Resolving packages with \`swift\`"
+# Resolve the packages using the correct method
+if [[ -d "${XCWORKSPACE_PATH}" ]]; then
+  echo "~~~ Resolving Swift Packages with \`xcodebuild\`"
+  if [[ -z "${SCHEME_NAME}" ]]; then
+    SCHEME_NAME=$(xcodebuild -workspace "${XCWORKSPACE_PATH}" -onlyUsePackageVersionsFromResolvedFile -skipPackageUpdates -list | sed -n '/Schemes:/{n;s/^ *//p;}')
+    echo "No \`--scheme\` parameter provided; assuming the first one found (as it shouldn't matter in practice?): \`${SCHEME_NAME}\`"
```

Comment:
Suggesting to remove the "as it shouldn't matter in practice?" parenthetic because, while it makes sense as an observation in the implementation (and as far as I have seen it is correct) I don't feel it adds value to the logs. The caller didn't pass a scheme, so they'll have to deal with whichever one we default to for them.

```suggestion
    echo "No \`--scheme\` parameter provided. Using the first one found in the list: \`${SCHEME_NAME}\`"
```